runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9092,8 +9099,15 @@ J9::Z::TreeEvaluator::VMnewEvaluator(TR::Node * node, TR::CodeGenerator * cg)
             {
             classSymRef = node->getSecondChild()->getSymbolReference();
             reloKind = TR_VerifyRefArrayForAlloc;
-            }
 
+            if (comp->getOption(TR_UseSymbolValidationManager))
+               classToValidate = comp->fej9()->getComponentClassFromArrayClass(classToValidate);
```

Comment:
Might be worth having a comment here stating why this is necessary. I don't think there was a comment on the x codegen because I dropped the ball there :).

The reason we need to validate the component class is because in AOT w/o SVM, we validate the class by pulling it out of the constant pool; however that class is not the array class (because the `anewarray` bytecode refers to the component class). However, in the evaluator, we directly refer to the array class. In order to have the relo infrastructure be consistent, we need to remember to validate the component class (since the relo infrastructure is expecting the component class).